darts/tests/models/forecasting/test_torch_forecasting_model.py

```diff
@@ -471,6 +471,31 @@ def test_load_weights(self):
                 f"respectively {retrained_mape} and {original_mape}",
             )
 
+        def test_resume_training_from_checkpoint(self):
+            model = RNNModel(
+                12,
+                "RNN",
+                10,
+                10,
+                n_epochs=10,
+                work_dir=self.temp_work_dir,
+                save_checkpoints=True,
+                model_name="test_rnn",
+                force_reset=True,
+            )
+
+            model.fit(self.series)
+            self.assertEqual(10, model.epochs_trained)
+
+            model_from_checkpoint = RNNModel.load_from_checkpoint(
+                "test_rnn", work_dir=self.temp_work_dir, best=False
+            )
+            # epochs_trained is still 10 when loading from checkpoint
+            self.assertEqual(10, model_from_checkpoint.epochs_trained)
+
+            model_from_checkpoint.fit(self.series, epochs=15)
```

Comment:
I think that in order to check if the issue is properly fixed, the `epochs` argument should be equal to 5 (smaller than the value of `epochs_trained` in the loaded checkpoint, which was causing issue when loaded by the PytorchLightning trainer).